lib/kb_ReadsUtilities/kb_ReadsUtilitiesImpl.py

```diff
@@ -81,6 +79,46 @@ def log(self, target, message):
         print(message)
         sys.stdout.flush()
 
+    def add_id_to_plus_line(self,readsfile):
+        gzip_flag = False
+        if readsfile.lower().endswith('.gz'):
+            gzip_flag = True
+
+        if gzip_flag:
+            f = gzip.open(readsfile, 'rt')
+        else:
+            f = open(readsfile, 'r')        
```

Comment:
It's good to use `with` for file handling. File might left open.
`with (gzip.open if gzip_flag else open)(readsfile, 'rt') as f`

Reply:
done